### ISSUE TITLE
[syncfusion_flutter_pdf] selection of 2nd line doesn't work (unless you first move to the last line)

### ISSUE THREAD

**Issue by MarcVanDaele90** [syncfusion/flutter-widgets#1646]
I'm testing on Flutter-web, using   syncfusion_flutter_pdfviewer: ^24.1.47

I've tried the example https://pub.dev/packages/syncfusion_flutter_pdf/example but I used the pdf locally from assets
```
      body: SfPdfViewer.asset("assets/twopages.pdf",
        key: _pdfViewerKey,
      ),

```
where twopages.pdf is [this example file ](https://www.africau.edu/images/default/sample.pdf)

My index.html contains
```
<script src="//cdnjs.cloudflare.com/ajax/libs/pdf.js/2.11.338/pdf.min.js"></script>
<script type="text/javascript">
   pdfjsLib.GlobalWorkerOptions.workerSrc = "//cdnjs.cloudflare.com/ajax/libs/pdf.js/2.11.338/pdf.worker.min.js";
</script>
```

I have an issue with selecting text, especially when I want to select multiple lines of text.
When I start the selection at the first row, the selection starts fine but when I move my mouse to the line(s) below, the selection does not extend beyond the first line.  Only when why mouse pointer reaches the last line, all lines get selected.

Pleas let me know if you can reproduce this issue and/or whether you need additional information.

**Comment by immankumarsync** [comment]
Hi @MarcVanDaele90,
We are unable to replicate the issue and the text selection in multiple lines is working as expected. Please find the attached video

Could you please check multiple line text selection with different pdf documents?

https://github.com/syncfusion/flutter-widgets/assets/93641486/2e4b7dd5-a25a-4882-93f0-edc7b3ed2be6

**Comment by MarcVanDaele90** [comment]
Hi @immankumarsync  Thanks for your prompt reply!
I see this behavior in multiple files. I've attached a recording of what I see on my side.
Moving the cursor down does not select additional lines, unless you reach the last line.
Moving the cursor upwards again behaves as expected.

I see the issue both on localhost and when published to a (test)webserver.  After publishing, the same behavior was observed from Chrome on a Windows PC.

https://github.com/syncfusion/flutter-widgets/assets/90178346/68fa3586-5977-4ac2-940b-d06543779748

The output of `flutter doctor -v` is below

Looking forward to your feedback!
```
[✓] Flutter (Channel stable, 3.16.5, on Linux Mint 20.2 5.15.0-56-generic,
    locale en_US.UTF-8)
    • Flutter version 3.16.5 on channel stable at
      /home/marc/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 78666c8dc5 (6 weeks ago), 2023-12-19 16:14:14 -0800
    • Engine revision 3f3e560236
    • Dart version 3.2.3
    • DevTools version 2.28.4

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /home/marc/Android/Sdk
    • Platform android-TiramisuPrivacySandbox, build-tools 33.0.0
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.17+8-post-Ubuntu-1ubuntu220.04)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[!] Android Studio (version 2022.1)
    • Android Studio at /home/marc/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.74.0)
    • VS Code at /usr/share/code
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Linux Mint 20.2 5.15.0-56-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 120.0.6099.216

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

```

**Comment by immankumarsync** [comment]
Hi @MarcVanDaele90, we have checked with multiple pdf documents and multiline selection is working properly on our side. We suspect that the issue might be specific to pdf documents. Could you please share the documents in which you are facing the issue?

**Comment by MarcVanDaele90** [comment]
It indeed works fine on quite some files.

I see the reported issue in 
[example3.pdf](https://github.com/syncfusion/flutter-widgets/files/14103299/example3.pdf)

I see a similar(?) issue in 
[example2.pdf](https://github.com/syncfusion/flutter-widgets/files/14103318/example2.pdf) where text selection stops at some point (see image) and the selection doesn't extend beyond "toevalstreffer"
![image](https://github.com/syncfusion/flutter-widgets/assets/90178346/e7b70776-13f7-418a-a963-668894dc41c8)
This file also reports a lot of errors like
`RangeError (index): Index out of range: index should be less than 14: 14`
when I hit this point (so this might be a different issue)
I have another file that also shows this RangeError.

Hope this helps

**Comment by MarcVanDaele90** [comment]
Hi @immankumarsync  Can you reproduce the issue(s) with the two given files?

**Comment by immankumarsync** [comment]
Hi @MarcVanDaele90, we are able to replicate the issue, "Text selection is not working in specific PDF documents". Currently we are validating the issue, and we will update further details on February 6, 2024.

**Comment by immankumarsync** [comment]
@MarcVanDaele90, we confirmed the issue "Extracting text lines provide incorrect results for specific PDF documents" as a defect in our product and fix will be included in our upcoming weekly release, which will be available on February 13, 2024.

**Comment by immankumarsync** [comment]
@MarcVanDaele90, the issue "Text selection is not working in specific PDF documents" has been resolved, and the fix for the same is available with the latest version [24.2.5](https://pub.dev/packages/syncfusion_flutter_pdfviewer/versions/24.2.5).

**Root cause:**
* In example2.pdf, joint characters in the word 'toevalstreffer' cause the word and glyph length to mismatch, causing a range error while text selection.
* In example3.pdf, the textline bounds was not calculated properly.
